### ISSUE TITLE
Compatibility with Composer 2

### ISSUE THREAD

**Issue by gbiorczyk** [wp-cli/package-command#125]
Hi,

Hello everyone. Thanks for your work, I appreciate it very much.

I would like to adapt my composer.json to Composer 2. For testing purposes I have created a very simple composer.json file:
```
{
  "name": "gbiorczyk/my-package",
  "require-dev": {
    "composer/composer": "^2.0"
  }
}
```

And then I execute the following command:
```
composer require wp-cli/package-command
```

And I have such errors:
```
sing version ^2.0 for wp-cli/package-command
./composer.json has been updated
Running composer update wp-cli/package-command
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - wp-cli/package-command v2.0.0 requires composer/composer ^1.2.0 -> found composer/composer[1.2.0, ..., 1.10.17] but it conflicts with your root composer.json require (^2.0).
    - wp-cli/package-command[v2.0.1, ..., v2.0.7] require composer/composer >=1.2.0 <1.7.0 || ^1.7.1 -> found composer/composer[1.2.0, ..., 1.10.17] but it conflicts with your root composer.json require (^2.0).
    - Root composer.json requires wp-cli/package-command ^2.0 -> satisfiable by wp-cli/package-command[v2.0.0, ..., v2.0.7].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

Do you plan on compatibility with Composer 2? If so, when can this be expected? I will be very grateful for your answer!

With best wishes,
Mateusz Gbiorczyk

**Comment by gilzow** [comment]
@gbiorczyk  for what it's worth, there are additional packages required by wp-cli that are not composer2 compatible. The team behind wp-cli will need to ensure that all of the packages required by wp-cli-bundle https://github.com/wp-cli/wp-cli-bundle/blob/master/composer.json are updated to be compatible with Composer2

**Comment by gbiorczyk** [comment]
Do you know anything more? Do you know any possible dates? Anyone working on this?

@gilzow, I will be very grateful for any additional information.

**Comment by schlessera** [comment]
I'm currently working on this and hope to have it work with Composer v2 at the end of the week. No promises, though, as I haven't gone through all the bugs that need to be solved.

**Comment by gilzow** [comment]
@gbiorczyk I forked both wp-cli-bundle and package-command packages and updated both to work with Composer2, and then pointed my projects to my forked version. Upon build, it failed on the next required package from wp-cli-bundle. At that point, I decided I didnt want to fork all the other packages.

**Comment by schlessera** [comment]
Fixed via #127.

@gbiorczyk, @gilzow  Can you confirm it now works as expected for your use case?

**Comment by gbiorczyk** [comment]
@schlessera Thank you very much! I see you need to use minimum-stability at `dev` level.

Do you have any plans for this to work on `stable` level?

**Comment by schlessera** [comment]
@gbiorczyk: Right now, it pulls in `wp-cli/wp-cli` at `dev-master`, because it requires a change that that repo has no release for yet. This will be solved before WP-CLI v2.5.0 is finalized, which should be next week.

See https://github.com/wp-cli/package-command/issues/118
